src/mark-suspicious-activity-as-reported.ts

```diff
@@ -0,0 +1,109 @@
+import { SNSEvent } from "aws-lambda";
+import {
+  SendMessageCommand,
+  SendMessageCommandOutput,
+  SendMessageRequest,
+  SQSClient,
+} from "@aws-sdk/client-sqs";
+import {
+  DynamoDBDocumentClient,
+  UpdateCommand,
+  QueryCommand,
+} from "@aws-sdk/lib-dynamodb";
+import { DynamoDBClient } from "@aws-sdk/client-dynamodb";
+import { ActivityLogEntry } from "./common/model";
+
+const dynamoClient = new DynamoDBClient({});
+const dynamoDocClient = DynamoDBDocumentClient.from(dynamoClient);
+
+export const sendSqsMessage = async (
+  messageBody: string,
+  queueUrl: string | undefined
+): Promise<SendMessageCommandOutput> => {
+  const { AWS_REGION } = process.env;
+  const client = new SQSClient({ region: AWS_REGION });
+  const message: SendMessageRequest = {
+    QueueUrl: queueUrl,
+    MessageBody: messageBody,
+  };
+  return client.send(new SendMessageCommand(message));
+};
+
+export const getItemByEventId = async (
+  tableName: string,
+  indexName: string,
+  eventId: string
+): Promise<{ user_id: string; timestamp: number }> => {
+  const getItem = new QueryCommand({
+    TableName: tableName,
+    IndexName: indexName,
+    KeyConditionExpression: "event_id = :event_id",
+    ExpressionAttributeValues: {
+      ":event_id": eventId,
+    },
+  });
+
+  const result = await dynamoDocClient.send(getItem);
+
+  if (result?.Items?.length !== 1) {
+    throw Error(
+      `Expecting exactly 1 result from getItemByEventId, but got ${result?.Items?.length}`
```

Comment:
We know we have a length defined at this point.
```suggestion
      `Expecting exactly 1 result from getItemByEventId, but got ${result.Items.length}`
```

Reply:
`Items` is an optional field, so potentially can be `undefined`, so the optional chaining here stops a typescript error

Reply:
You're right, Items can still be undefined all we know is that it isn't 1.